pydicom/config.py

```diff
@@ -152,14 +153,79 @@ def DS_decimal(use_Decimal_boolean: bool = True) -> None:
 Default ``False``.
 """
 
+
 enforce_valid_values = False
-"""Raise exceptions if any value is not allowed by DICOM Standard.
+"""Obsolete. Use settings.reading_validation_mode instead."""
 
-e.g. DS strings that are longer than 16 characters; IS strings outside
-the allowed range.
 
-Default ``False``.
-"""
+# Constants used to define how data element values shall be validated
+NO_VALIDATION = 0
+"""No value validation will be performed."""
+
+WARN_ON_ERROR = 1
+"""A warning is issued on a value validation error."""
+
+RAISE_ON_ERROR = 2
+"""An exception is raised on a value validation error."""
+
+
+class Settings:
+    """Collection of several configuration values.
+
+    Attributes
+    ----------
+    reading_validation_mode : int
+        Defines behavior for value validation while reading value.
+        Value validation checks if a value is allowed by the DICOM Standard,
+        e.g. that DS strings are not longer than 16 characters and contain only
+        allowed characters.
+        The default (WARN_ON_ERROR) is to log a warning in the case of an
+        invalid value, RAISE_ON_ERROR will raise an error in this case, and
+        NO_VALIDATION will bypass the validation.
+    writing_validation_mode : int
+        Defines behavior for value validation while writing a value.
+        See :attr:`~pydicom.config.Settings.reading_validation_mode`.
+    """
+
+    def __init__(self) -> None:
+        self._reading_validation_mode: int = WARN_ON_ERROR
+        self.writing_validation_mode = RAISE_ON_ERROR
```

Comment:
I'm going back and forth with how to define this settings, I'm still not sure what the correct way is. I prefer to have the settings in a class, as this makes it possible to change them later using properties, while for module-level attributes this is not possible.